### ISSUE TITLE
masking/types.d.ts imports from "./internal/types.ts"

### ISSUE THREAD

**Issue by Yogu** [apollographql/apollo-client#12170]
### Issue Description

Starting in @apolllo/client@3.12.0, the file `masking/types.d.ts` has this import statement:

```
import type { ContainsFragmentsRefs, IsAny, RemoveFragmentName, RemoveMaskedMarker, UnwrapFragmentRefs } from "./internal/types.ts";
```

The file `types.ts` does not exist in the NPM package. The file `types.d.ts` would exist.

This results in this typescript error

```
Error: node_modules/@apollo/client/masking/types.d.ts:2:111 - error TS2691: An import path cannot end with a '.ts' extension. Consider importing './internal/types.js' instead.

2 import type { ContainsFragmentsRefs, IsAny, RemoveFragmentName, RemoveMaskedMarker, UnwrapFragmentRefs } from "./internal/types.ts";
                                                                                                                ~~~~~~~~~~~~~~~~~~~~~
```

The fix is probably to change [this line](https://github.com/apollographql/apollo-client/blob/36b93b905f8e073e94ef12aaa89ca27451eb3e85/src/masking/types.ts#L8) to import from `.js` instead of `.ts`.

### Link to Reproduction

Let me know if you need this

### `@apollo/client` version

3.12.0

**Comment by phryneas** [comment]
Hi @Yogu,
that's a great catch!

Could you try the CI build from #12171 to see if it fixes the problem?

**Comment by jerelmiller** [comment]
We just released 3.12.1 which should fix this. Thanks for raising the issue!

**Comment by Yogu** [comment]
That was quick, thank you! I just tested 3.12.1 and it seems to fix the issue.
